cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/OptimizerResult.java

```diff
@@ -194,6 +212,14 @@ public boolean isCapacityEstimated() {
     return _optimizationOptions.excludedBrokersForReplicaMove();
   }
 
+  /**
+   * @return The string describing goal result.
+   */
+  public String goalResultDescription(String goalName) {
+    return _violatedGoalNamesBeforeOptimization.contains(goalName) ?
```

Comment:
personally I think this chain of ternary is very hard to read/debug. please consider expanding this to if/else instead.